src/Marten/IRetryPolicy.cs

```diff
@@ -66,4 +67,115 @@ public Task<TResult> ExecuteAsync<TResult>(Func<Task<TResult>> operation, Cancel
             return operation();
         }
     }
+
+    /// <summary>
+    /// Default retry policy, which accounts for <see cref="NpgsqlException.IsTransient"/>.
+    /// </summary>
+    /// <remarks>
+    /// Based on example https://martendb.io/documentation/documents/advanced/retrypolicy/ by Joona-Pekka Kokko.
+    /// </remarks>
+    public sealed class DefaultRetryPolicy: IRetryPolicy
+    {
+        private readonly int _maxTries;
+        private readonly Func<Exception, bool> _filter;
+        private readonly Func<int, TimeSpan> _sleep;
+
+        private static readonly Func<Exception, bool> DefaultFilter = x => x is NpgsqlException npgsqlException && npgsqlException.IsTransient;
+        private static readonly Func<int, TimeSpan> DefaultSleep = x => TimeSpan.FromSeconds(x);
+
+        private DefaultRetryPolicy(int maxTries, Func<Exception, bool> filter, Func<int, TimeSpan> sleep)
+        {
+            _maxTries = maxTries;
+            _filter = filter ?? DefaultFilter;
+            _sleep = sleep ?? DefaultSleep;
+        }
+
+        /// <summary>
+        /// Initializes a retry policy that will retry once after failure.
+        /// </summary>
+        /// <param name="filter">Optional filter when to apply, default to checking for <see cref="NpgsqlException.IsTransient"/></param>
+        /// <param name="sleep">Optional sleep after exception, gets retry number 1-N as parameter, defaults to sleeping retry number seconds</param>
+        /// <returns>The configured retry policy.</returns>
+        public static IRetryPolicy Once(Func<Exception, bool> filter = null, Func<int, TimeSpan> sleep = null)
```

Comment:
**KUDOS** I really like what you did with Once, Twice and Times. Nice API design 👍

Reply:
Can't take the credit, much based on @jokokko 's work 🙂